core/src/utility/console.cc

```diff
@@ -0,0 +1,138 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2017 Serge Zaitsev
+ * Copyright (c) 2022 Steffen André Langnes
+ * Copyright (c) 2025 Michael Jonker
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#pragma once
+#if defined(__cplusplus) && !defined(WEBVIEW_HEADER)
+
+#include "webview/utility/console.hh"
+#include <iostream>
+
+using namespace webview::utility;
+
+void console::warn(std::string message) {
+  init_console();
+  std::string prefix = "[WEBVIEW][WARNING]: ";
+  message = set_colour(colours.yellow, prefix + message);
+  printf("%s\n", message.c_str());
+  free_console();
+};
+
+void console::error(std::string message) {
+  init_console();
+  std::string prefix = "[WEBVIEW][ERROR]: ";
+  message = set_colour(colours.red, prefix + message);
+  std::cerr << message << "\n";
+  free_console();
+};
+
+std::string console::set_colour(int color, std::string message) {
+  if (!is_console_mode_set) {
+    return message;
+  };
+  message = "\x1B[" + std::to_string(color) + "m" + message + "\033[0m";
+  return message;
+}
+
+colours_t console::colours{33, 31};
+
+#if defined(_WIN32)
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+#ifdef _MSC_VER
+#define _WIN32_WINNT 0x0601
```

Comment:
MSVC compiler wants a windows version, else fails.

This is Windows min version 7